enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,1319 @@
+
+- What happens when a cluster-admin adds invalid EIPs ?
+
+  The load balancer creation would not proceed and fail as per details mentioned in [subsection of Test Plan](#Test when Ingress Controller CR has been provided with invalid EIPAllocations.)
+
+- What happens when a cluster-admin adds number of EIPs which are not equal to the number of public subnets for the VPC id tagged with cluster name.
+
+  The load balancer creation would not proceed and fail as per details mentioned in [subsection of Test Plan](#test-when-ingress-controller-cr-has-been-provided-with-number-of-eips-not-equal-to-the-number-of-public-subnets-for-vpc-tagged-with-your-cluster-name)
+
+- What happens when a cluster-admin adds nothing under eipAllocations ?
+  
+  The load balancer would be created with no eipAllocations and the service annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` will be removed.
+  For more check [this](#Cluster admin wants operator to update the Load Balancer service for updated Ingress Controller eipAllocations)
+  The NLB will get the regular IP address assigned in that case.
+
+## Test Plan
+
+### This EP will be covered by unit tests in the following:
+- cluster-ingress-operator
+- installer
+
+### This  EP will also be covered by end to end tests.
+##### Create an Ingress Controller using Ingress Controller CR having eipAllocations.
+  The end to end test will cover the scenario where user will set an `eipAllocations` field
+  in the Ingress Controller CR and then test if the Ingress Controller
+  with service type load balancer with the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+  has value of `eipAllocations` field from the Ingress Controller CR was created.
+  We will check the status of IngressController controller for LoadBalancer which is `LoadBalacerReady`, `LoadBalancerProgressing`, `IngressControllerUnavailable`
+  and any errors in the cloud controller manager logs.
+
+##### Update an Ingress Controller eipAllocations using Ingress Controller CR which already has eipAllocations set.
+  The end to end test will also the cover the scenario where user updates the `eipAllocations` field
+  in the Ingress Controller CR and then test if the Ingress Controller
+  with service type load balancer with the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`
+  has value of `eipAllocations` field from the Ingress Controller CR was updated.
+  We will check the status of IngressController controller for LoadBalancer which is `LoadBalacerReady` and `LoadBalancerProgressing`
+  and the cloud controller manager logs.
+  The `LoadBalancerProgressing` should display `False` and message `"'One or more status conditions indicate progressing: LoadBalancerProgressing=True
```

Comment:
Sorry this doesn't make sense to me. `LoadBalancerProgressing` should display `False`, and then message with `LoadBalancerProgressing=True`? Is that a mistake?

Also, this E2E test doesn't talk about deleting the service. The flow doesn't make sense, you say 1. Update `eipAllocations` 2. Check service annotation 3. Check LoadBalancerProgressing.

The service annotation isn't going to change if you haven't deleted the service.

It feels like there's a lot of fluff, but it's missing some key parts. I think you could sum this up in one sentence:
```
Update an IngressController with new `eipAllocations`, observe `LoadBalancerProgressing=True`,
deleting the LoadBalancer-type service, and observe value of
`service.beta.kubernetes.io/aws-load-balancer-subnets`.
```

Feel free to use my E2E tests section as reference: https://github.com/openshift/enhancements/pull/1595/files#diff-f8f6da2366b6e9f9eede942d6a878c8043690acb03727fad6bbf71d37c95fd92R475-R487

Reply:
> Sorry this doesn't make sense to me. `LoadBalancerProgressing` should display `False`, and then message with `LoadBalancerProgressing=True`? Is that a mistake?

Yes, I will correct it.

Reply:
Actually, after update of eip, the e2e test will just check for status message instructing the admin to manually delete the service.

I will correct it. Thanks for pointing out.

Automatic deletion will happen with auto-delete-loadbalancer annotation for which an e2e will be written.